Lesson15/Lesson15/gokce/script.js

```diff
@@ -0,0 +1,391 @@
+*/
+function isPalindrome(str){
+  let strPalindrome=str.split("").reverse("").join("");
+  if(str===strPalindrome){
+    console.log("<str> is a palindrome")
+  }else{
+    console.log("<str> is not a palindrome");
+  }
+}
+isPalindrome("gokce");
+isPalindrome("level");
+
+/*
+10. Check Even or Odd (if-else)
+   - Define a function `evenOrOdd(number)` that:
+     - Logs "Even" if the number is even
+     - Logs "Odd" if the number is odd
+*/
+function evenOrOdd(number){
+  let numberCheck=number%2;
+  if(numberCheck===0){
+    console.log("Even");
+  }else{
+    console.log("Odd");
+  }
+}
+evenOrOdd(5);
+evenOrOdd(6);
+
+/*
+11. URL Protocol Checker
+   - Define a function `checkProtocol(url)` that converts the URL to lowercase
+     and checks if it starts with "https" using .startsWith().
+   - Log "Secure connection" if true, otherwise "Unsecure connection".
+*/
+function checkProtocol(url){
+const urlLower=url.toLowerCase();
+if(urlLower.startsWith("https")){
+  console.log("Secure connection");
+}else{
+  console.log("Unsecure connection");}
+}
+checkProtocol("https://www.google.com");
+checkProtocol("http://www.google.com");
+
+/*
+13. Repeat a String
+   - Define a function `repeatWord(word, times)` that uses the .repeat() method
+     to repeat `word` `times` times.
+   - Log the repeated result.
+*/
+function repeatWord(word,times){
+  console.log(word.repeat(times));
+}
+repeatWord("gokce",3);
+repeatWord("gokce",2.5);
+/*
+14. Replace Substring
+   - Define a function `censorWord(sentence, target)` that replaces `target`
+     with "****" (use .replaceAll() or multiple .replace()).
+   - Log the censored sentence.
+*/
+function censorWord(sentence,target){
+  let censorWord=[];
+  let newSentence=sentence.replaceAll(target,function(match){censorWord.push(match);
+    return "****";});
+  
+  console.log("new sentence:", newSentence);
+  console.log("censored word:", censorWord);
+}
+censorWord("gokce is a student","gokce");
+/*
+15. Check First Character (if-else)
+   - Define a function `startsWithA(str)` that checks if the string starts with 'A'
+     (use .charAt(0) or [0]).
+   - Log "Starts with A" or "Does not start with A".
+*/
+function startsWithA(str){
+  const controlString=str.charAt(0);
+  if (controlString==="A"){
+    console.log("Starts with A");
+  }else{
+    console.log("Does not start with A");
+  }
+}
+startsWithA("Alignment");
+/*
+16. Slice Last N Characters
+   - Define a function `sliceLastN(text, n)` that uses .slice(-n) to extract
+     the last `n` characters of `text`.
+   - Log the result.
+*/
+function sliceLast(text,n){
+  console.log(text.slice(-n));
+}
+sliceLast("Technology",2);
+sliceLast("Develop",3);
+
+/*
+17. Switch: Grade Checker
+   - Define a function `gradeChecker(score)` that uses a switch (or if-else chain):
+     90+ -> "A"
+     80-89 -> "B"
+     70-79 -> "C"
+     60-69 -> "D"
+     below 60 -> "F"
+   - Log the grade.
+*/
+function gradeChecker(score){
+  let grade="";
```

Comment:
IN this case, you normally don't need a variable. You could write your function like this instead:

```
function checkGrade(score){
  if (score >= 90) {
    return "A";
  } else if (score>=80) {
    return "B";
  }
....
}


```